ruby/example_code/s3/tests/test_auth_federation_token_request_test.rb

```diff
@@ -0,0 +1,14 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX - License - Identifier: Apache - 2.0
+
+require_relative '../auth_federation_token_request_test.rb'
+require_relative './s3_test_globals.rb'
```

Comment:
if you can avoid globals here, you should do so.. they're discouraged. you can use `let()` blocks to define test data

```
let(:region) { 'us-west-2' }
let(:bucket_name) { 'my_bucket_name' }

let(:s3_client) { Aws::S3::Client.new(stub_responses: true) }
..etc
```

Reply:
I switched from globals to let blocks, as you propose.